exonum-java-binding/core/src/test/java/com/exonum/binding/runtime/Pf4jServiceLoaderTest.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.runtime;
+
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.exonum.binding.service.AbstractServiceModule;
+import com.exonum.binding.service.ServiceModule;
+import com.google.common.collect.ImmutableList;
+import java.net.URI;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.pf4j.PluginManager;
+import org.pf4j.PluginState;
+
+@SuppressWarnings("WeakerAccess")
+@ExtendWith(MockitoExtension.class)
+class Pf4jServiceLoaderTest {
+
+  @Mock
+  private PluginManager pluginManager;
+
+  @InjectMocks
+  private Pf4jServiceLoader serviceLoader;
+
+  @Test
+  void canLoadService() throws ServiceLoadingException {
+    URI artifactLocation = URI.create("file:///tmp/service.jar");
```

Comment:
should we use the same format in the console for `--ejb-artifact-uri` param?

Reply:
No, absolute paths must work. Can actually extract that in a constant.

Reply:
`URI.create("/tmp/");` leads to `java.lang.IllegalArgumentException: Missing scheme`

Reply:
What's your Java version? 
```
jshell> URI.create("/tmp/foo.jar");
$1 ==> /tmp/foo.jar

jshell> URI.create("/tmp/");
$2 ==> /tmp/
```

Reply:
sorry, it fails on `Paths.get(uri)` with the IAE:
```
  @Test
  void test() {
    URI uri = URI.create("/tmp/");
    Path path = Paths.get(uri);
    System.out.println(path);
  }
```

Reply:
Thanks! I wonder if we shall just use paths. Even if the DS supports various protocols, we can push the responsibility of downloading the artifacts and storing in the local FS on the core 🤔